### PR TITLE
Revert removal of 'TestReporter'

### PR DESCRIPTION
**Pull request by jvandort**
This was used by cashapp/paparazzi

Fixes https://github.com/gradle/gradle/issues/31783

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test rfr

File: testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/PaparazziPluginSmokeTest.groovy [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        file("gradle.properties") << """
+            android.useAndroidX = true
+        """
+
+        file("src/main/AndroidManifest.xml") << """<?xml version="1.0" encoding="utf-8"?>
+            <manifest xmlns:android="http://schemas.android.com/apk/res/android">
+            </manifest>""".stripIndent()
+
+        expect:
+        agpRunner(agpVersion, 'testDebug').build()
```
Comment (big-guy) [inline testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/PaparazziPluginSmokeTest.groovy:68]: Doesn't this need a test or something to really exercise what it's doing?

**Review by big-guy — APPROVED** [review]
+1 on adding the smoke test coverage. I'm not sure if we exercise it enough to know if we broke something though?

**Comment by jvandort** [comment]
I did verify this test fails before the revert

File: testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/PaparazziPluginSmokeTest.groovy (hunk 1 above)
Comment (jvandort) [reply testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/PaparazziPluginSmokeTest.groovy:68]: It's a smoke test. No exception means no smoke!

We could probably be a bit more comprehensive here but for the time constraints, I felt this was adequate for now

**Pull request merged**